pkg/cmd/workflow/run/run.go

```diff
@@ -0,0 +1,382 @@
+
+	return cmd
+}
+
+type InputAnswer struct {
+	providedInputs map[string]string
+}
+
+func (ia *InputAnswer) WriteAnswer(name string, value interface{}) error {
+	if s, ok := value.(string); ok {
+		ia.providedInputs[name] = s
+		return nil
+	}
+
+	// TODO i hate this; this is to make tests work:
+	if rv, ok := value.(reflect.Value); ok {
+		ia.providedInputs[name] = rv.String()
+		return nil
+	}
+
+	return fmt.Errorf("unexpected value type: %v", value)
+}
+
+func runRun(opts *RunOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not build http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	states := []shared.WorkflowState{shared.Active}
+	workflow, err := shared.ResolveWorkflow(
+		opts.IO, client, repo, opts.Prompt, opts.Selector, states)
+	if err != nil {
+		var fae shared.FilteredAllError
+		if errors.As(err, &fae) {
+			return errors.New("no workflows are enabled on this repository")
+		}
+		return err
+	}
+
+	ref := opts.Ref
+
+	if ref == "" {
+		ref, err = api.RepoDefaultBranch(client, repo)
+		if err != nil {
+			return fmt.Errorf("unable to determine default branch for %s: %w", ghrepo.FullName(repo), err)
+		}
+	}
+
+	yamlContent, err := getWorkflowContent(client, repo, workflow, ref)
+	if err != nil {
+		return fmt.Errorf("unable to fetch workflow file content: %w", err)
+	}
+
+	inputs, err := findInputs(yamlContent)
+	if err != nil {
+		return err
+	}
+
+	providedInputs := map[string]string{}
+
+	if opts.Prompt {
+		qs := []*survey.Question{}
+		for inputName, input := range inputs {
+			q := &survey.Question{
+				Name: inputName,
+				Prompt: &survey.Input{
+					Message: inputName,
+					Default: input.Default,
+				},
+			}
+			if input.Required {
+				q.Validate = survey.Required
+			}
+			qs = append(qs, q)
+		}
+
+		sort.Slice(qs, func(i, j int) bool {
+			return qs[i].Name < qs[j].Name
+		})
+
+		inputAnswer := InputAnswer{
+			providedInputs: providedInputs,
+		}
+		err = prompt.SurveyAsk(qs, &inputAnswer)
+		if err != nil {
+			return err
+		}
+	} else {
+		if opts.JSON != "" {
+			err := json.Unmarshal([]byte(opts.JSON), &providedInputs)
+			if err != nil {
+				return fmt.Errorf("could not parse provided JSON: %w", err)
+			}
+		}
+
+		if len(opts.InputArgs) > 0 {
+			fs := pflag.FlagSet{}
+			for inputName, input := range inputs {
+				fs.String(inputName, input.Default, input.Description)
+			}
+			err = fs.Parse(opts.InputArgs)
+			if err != nil {
+				return fmt.Errorf("could not parse input args: %w", err)
+			}
+			for inputName := range inputs {
+				providedValue, _ := fs.GetString(inputName)
+				providedInputs[inputName] = providedValue
+			}
+		}
+	}
+
+	for inputName, inputValue := range inputs {
+		if inputValue.Required && providedInputs[inputName] == "" {
```

Comment:
The input might be marked as "required", but if it has a default value then it's not required to be filled in by the user.